src/main/java/org/checkerframework/specimin/UnsolvedSymbolVisitor.java

```diff
@@ -176,6 +178,16 @@ public class UnsolvedSymbolVisitor extends ModifierVisitor<Void> {
   /** New files that should be added to the list of target files for the next iteration. */
   private final Set<String> addedTargetFiles = new HashSet<>();
 
+  /** Stores the sets of method names declared in the currently visiting classes. */
+  private final ArrayDeque<Set<MethodDeclaration>> declaredMethod = new ArrayDeque<>();
+
+  /**
+   * Maps the name of a class to the unsolved interface that it implements. If a class implements
+   * multiple unsolved interfaces, only the last one is tracked in this map.
```

Comment:
This makes me nervous, for a reason that I can't really articulate. I have a suspicion that at some point in the future one of us will come back to this comment and wonder "why didn't we just store the whole list of interfaces? that would be really convenient for whatever task I'm trying to do now".

I think it would therefore be better for maintainability if we store the full list of interfaces here for each class, even if the current implementation only _uses_ the last one. The size of the data structure won't get much larger, and the code won't be all that much more complex. But, I think the resulting data structure will be less surprising to future maintainers (i.e., us in a few months).